numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -3628,11 +3672,19 @@ PyArray_GetDTypeTransferFunction(int aligned,
             }
         }
 
-        /* The special types, which have no byte-order */
+        /* The special types, which have no or subelement byte-order */
         switch (src_type_num) {
             case NPY_VOID:
-            case NPY_STRING:
             case NPY_UNICODE:
+                /* Wrap the copy swap function when swapping is necessary */
+                if (PyArray_ISNBO(src_dtype->byteorder) !=
+                        PyArray_ISNBO(dst_dtype->byteorder)) {
+                    return wrap_copy_swap_function(aligned,
+                                    src_stride, dst_stride,
+                                    src_dtype, 1,
+                                    out_stransfer, out_transferdata);
```

Comment:
Here too, I think we can call `PyArray_GetStridedZeroPadCopyFn` instead. 

That way, like before, `wrap_copy_swap_function` is only used for custom dtypes.